Qt: Detailed file browser table and other changes.

CozmoP opened the pull request:
## Description

- Detailed file browser table.
- Don't allow dropping files on file browser table.(it was adding the files to the selected playlist even if it was not visible, I found that confusing).
- Share selection between playlist views.
- Show grid footer on list view too.
- Add items count to footer.
- Rename a couple functions.
- Stretch playlist list view.
- Save grid view zoom level.
- Set fixed height to search and core selection docks to their minimum height.

## Reviewers

@bparker06 @Tatsuya79

twinaphex commented:
Seems there is a merge conflict that has to be resolved first.

Tatsuya79 commented:
Thanks for making the footer, that's really nice to switch between icon/list easily.
I would put a bit of spacing before the total items number though (just 1 char would be better).

Like the thumbnails associated with a playlist work with a folder of the same name in the file browser, core association did the same before.
But now if you go into a playlist, like Nintendo - NES associated to FCEUmm, then move to the file browser view and select a game in your Sega - Megadrive folder, the core association still shows FCEUmm.

That said, perhaps we need to think about people using folders without touching any playlist at all, and a way for them to do core associations.

CozmoP commented:
> 
> 
> Thanks for making the footer, that's really nice to switch between icon/list easily.
> I would put a bit of spacing before the total items number though (just 1 char would be better).
> 
I think that would be better adressed on the stylesheet.
>
> Like the thumbnails associated with a playlist work with a folder of the same name in the file browser, core association did the same before.
> But now if you go into a playlist, like Nintendo - NES associated to FCEUmm, then move to the file browser view and select a game in your Sega - Megadrive folder, the core association still shows FCEUmm.
>
I missed that. I pushed a dirty quick fix (hopefully) for now. Would need to be refactored.
I eventually want to make the grid view available for the file browser too, and show the thumbnails when available or the file icon otherwise.

Tatsuya79 commented:
Everything is on "Ask me" for core association in browser view here now.
Guess that's better.

Tatsuya79 commented:
Nice you're saving the zoom level now.

I would move "Icon view thumbnail type" out of menu>view>option to put it in the new footer as a drop down button similar to "View".

Tatsuya79 commented:
Just noticed that the History playlist is displaying the full path instead of the game name now.

Tatsuya79 commented:
Associations working fine in browser mode now, nice! :)

With that fancy browser it would be great to have an option to launch files with their associated OS app.
Something like: right click "Open with external program".

bparker06 commented:
Something strange is going on with long file names:

![img](https://i.imgur.com/QogdYR1.png)

Sometimes text that's too long is simply elided (which should be the correct behavior IMO), other times the text takes up multiple lines when there's no room to do so, leading to either non-centered text, text getting cut off or most of the name not even visible.

An even worse example:

![img](https://i.imgur.com/wnmzmFG.png)

I also noticed that the row numbers in list view are gone lately... was that intentional?

Tatsuya79 commented:
I don't that issue here on win7.
Did you try with a new retroarch_qt.cfg?

The row numbers were removed as they weren't following the new sorting method.

bparker06 commented:
Yes the same issue happens with a new retroarch_qt.cfg, this is on Linux (arch/xfce).

I tried on Windows and the issue does not happen there.

CozmoP commented:
I doesn't happen here on fedora 29 in either gnome wayland or xfce X11.
Using Qt 5.11.1

bparker06 commented:
5.12.0 on x11 here

CozmoP commented:
Please test again.
I also added the playlist views to a stacked widget to prevent a bug on some Qt version where hiding a table would advance its selection, as described [here](https://forum.qt.io/topic/48918/hiding-a-table-advances-the-selected-item).
Also renamed some things so they make more sense, and added the border to the file browser table.

Tatsuya79 commented on ui/drivers/qt/ui_qt_themes.h:
```diff
@@ -24,13 +24,16 @@ static const QString qt_theme_dark_stylesheet = QStringLiteral(R"(
       border-bottom:1px solid rgba(25,25,25,75%);
    }
+   QStackedWidget#centralWidget {
+      padding: 10px;
```
It should be 8px to be similar to the other borders.

CozmoP replied:
I changed the widgets that have a border to QFrame so that their follow [the box model](http://doc.qt.io/qt-5/stylesheet-customizing.html#the-box-model) and so the border can be changed in the stylesheet.
I also moved the playlist icon size there. My concern is that it may look too big on some screens so I don't know, maybe we should remove it.
In any case, it can be changed at runtime by loading a stylesheet.
